### ISSUE TITLE
Allow running `dotnet app.exe` instead of `dotnet app.dll`

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/sdk#22386]
When I target `net5.0` for an exe app, both `app.dll` and `app.exe` are published to output.

Running `dotnet app.dll` works fine, but running `dotnet app.exe` doesn't work:

```
C:\MetadataTools>dotnet C:\MetadataTools\src\BinaryCompatChecker\bin\Debug\net5.0\BinaryCompatChecker.exe
Error:
  An assembly specified in the application dependencies manifest (BinaryCompatChecker.deps.json) has already been found but with a different file extension:
    package: 'BinaryCompatChecker', version: '1.0.0'
    path: 'BinaryCompatChecker.dll'
    previously found assembly: 'C:\MetadataTools\src\BinaryCompatChecker\bin\Debug\net5.0\BinaryCompatChecker.exe'
```

This is not a great experience: we know what the user wants, let's just do the right thing.

The scenario is: an app is shipping both .dll and .exe. Some script runs the .exe. One should be able to prepend `dotnet` to run the .exe using .NET (by default it is run using Mono on Mac). Now I need to not only prepend `dotnet`, but also switch from `.exe` to `.dll`.

This is all very confusing for me, imagine how confusing it is for other users.

**Comment by agocke** [comment]
Nope, it's a native binary. The default deployment experience for the SDK (on any platform) is OS-specific. Portable apps do not use the apphost.

**Comment by marcpopMSFT** [comment]
Old issue triage: closing per Andy's comments above.
